[FEATURE] Permettre de renseigner la tarification de la part Pix dans l'import des candidats (PIX-4188).

clemlatz opened the pull request:
## :unicorn: Problème

Les pôles métiers en charge de la facturation des certifications payantes doivent pouvoir identifier simplement les candidats pour lesquels le centre de certification doit être facturé par Pix.

Aujourd’hui, il n’y a pas de moyen simple d’identifier les certifications à facturer au centre.

## :robot: Solution

Permettre aux centres de certification NON SCO (donc ne concerne ni le SCO isManagingStudent, ni le SCO isNOTManagingStudent !!!!) de renseigner la tarification pour chaque candidat lors de leur inscription en session de certification. Dans l’ods d’inscription des candidats :

### 1. Ajouter une colonne “Tarification part Pix”, OBLIGATOIRE, avec les options : 

- Gratuite
- Payante
- Prépayée

SI ce champ n’est pas rempli lors de l’import, afficher le message d’erreur suivant : 
> Ligne XX : Le champ “Tarification part Pix” est obligatoire.

### 2. ajouter une colonne “Code de prépaiement”, FACULTATIVE car ne doit être remplie QUE si l’option sélectionnée pour la “Tarification part Pix” pour un candidat donné est “Prépayée” : 

Ajouter un tooltip pour ce champ qui indique : 

> Code de prépaiement
> (Requis notamment dans le cas d'un achat de crédits combinés)
> Doit être composé du SIRET de l’organisation et du numéro de facture. Ex : 12345678912345/FACT12345
> Si vous ne possédez pas de facture, un code de prépaiement doit être établi avec Pix.

SI pour un candidat, “Prépayée” est sélectionné mais la colonne “Code de prépaiement” est vide, ALORS afficher le message d’erreur suivant lors de l’import : 
> Ligne XX : Le champ “Code de prépaiement” est obligatoire puisque l’option “Prépayée” a été sélectionnée pour ce candidat.

SI pour un candidat, un code de prépaiement a été renseigné, mais que l’option pour la “Tarification part Pix” est différente de “Prépayée”, ALORS afficher le message d’erreur suivant lors de l’import : 
> Ligne XX : Le champ “Code de prépaiement” doit rester vide puisque l’option “Prépayée” n'a pas été sélectionnée pour ce candidat.

SI pour un candidat, l’option “Prépayée” a été sélectionnée, et que le champ “Code de prépaiement est renseigné, ALORS le candidat est importé correctement.

⚠️ Ces deux nouvelles colonnes additionnelles sont à placer 
- après la colonne “Temps majoré ?” 
- avant les colonnes de certifications complémentaires 

## :rainbow: Remarques

*RAS*

## :100: Pour tester

*TODO*

clemlatz committed:
refactor(api): clarify code to add complementary certif. columns in ODS

clemlatz committed:
feat(api): add billing columns to certification-candidates table

clemlatz committed:
feat(api): add billing columns to CertificationCandidate model

pix-service commented:
I'm deploying this PR to these urls:

- App (.fr): https://app-pr3990.review.pix.fr
- App (.org): https://app-pr3990.review.pix.org
- Orga: https://orga-pr3990.review.pix.fr
- Certif: https://certif-pr3990.review.pix.fr
- Admin: https://admin-pr3990.review.pix.fr
- API: https://api-pr3990.review.pix.fr/api/

 Please check it out!

clemlatz committed:
feat(api): validate billing properties in CertificationCandidate model

clemlatz committed:
refactor(api): make code to add columns in ODS more generic

clemlatz committed:
refactor(api): make ODS processing code even clearer

Co-authored-by: Pierre Top <ptop@octo.com>

clemlatz committed:
fix(api): set correct width for column group header label

clemlatz committed:
fix(api): add missing covered-table-cells when using colspan

Covered table cells represent cells that are hidden because
they follow a cell with a colspan larger than 1. They are
necessary so that following cells are not hidden under the
ones with colspan larger than 1.

octo-topi committed:
Add billing columns

aceol committed:
:sparkles: feat(api) Retrieve SCO information for candidate import sheet

aceol committed:
:sparkles: feat(api) Hide billing columns for sco centers

octo-topi committed:
Feed actual candidate billing information when downloading template

octo-topi commented:
Remplacé par https://github.com/1024pix/pix/pull/4039
(on découpe en 2 PR)
